core/worker.go

```diff
@@ -509,21 +524,30 @@ func (w *worker) GeneratePendingHeader(block *types.WorkObject, fill bool) (*typ
 	atomic.StoreInt32(&w.newTxs, 0)
 
 	start := time.Now()
-	// Set the coinbase if the worker is running or it's required
-	var coinbase common.Address
-	if w.hc.NodeCtx() == common.ZONE_CTX && w.coinbase.Equal(common.Address{}) && w.hc.ProcessingState() {
-		w.logger.Error("Refusing to mine without etherbase")
+	// Set the primary coinbase if the worker is running or it's required
+	var primaryCoinbase common.Address
+	if w.hc.NodeCtx() == common.ZONE_CTX && w.primaryCoinbase.Equal(common.Address{}) && w.hc.ProcessingState() {
+		w.logger.Fatal("Refusing to mine without primaryCoinbase")
```

Comment:
Fatal seems a little excessive, what if someone just wants to run a node without mining?

Reply:
that is true, i guess i will change this back to Warn, but he wont be able to mine